src/app/beer_garden/scheduler.py

```diff
@@ -556,71 +546,29 @@ def create_job(job: Job) -> Job:
+                    rejected due to validation errors, along with the error message
+                    citing why it was rejected
     """
-    job_collection_name = "job"
-    id_key, oid_key = "_id", "$oid"
-
-    # we want the addition of Jobs to obey ACID: either all will succeed or none
-    # will succeed; hence we skip using mongoengine and use the pymongo library
-    # directly
-    the_connection: pymongo.MongoClient = mongoengine.get_connection()
-    the_db: pymongo.database.Database = mongoengine.get_db()
-
-    session: ClientSession
-    with the_connection.start_session() as session:
-        job_collection: Collection = the_db.get_collection(job_collection_name)
-
-        to_mongo_caller = methodcaller("to_mongo")
-        job_bson_objects = map(to_mongo_caller, map(from_brewtils, jobs))
-
-        # the below commented code does not work for mongodb running standalone;
-        # proper transactions are only supported when the database instance is a
-        # replica set or a distributed cluster
-        # (maybe one more point for getting rid of mongodb...)
-        #
-        # this is left here for reference in case anything changes; for example,
-        # a user actually running mongo in a replica set, in which case code to
-        # determine if that's the case can be added and then the following will
-        # work as intended:
-        #
-        # with session.start_transaction():
-        #     result: InsertManyResult = job_collection.insert_many(
-        #         job_bson_objects, session=session
-        #     )
-        #     job_ids_created_list = result.inserted_ids
-
-        result: InsertManyResult = job_collection.insert_many(
-            job_bson_objects, session=session
-        )
-        job_ids_created_list: List[ObjectId] = result.inserted_ids
-        jobs_created_cursor: Cursor = job_collection.find(
-            {id_key: {"$in": job_ids_created_list}}
-        )
-
-    # if we've reached this point, job_ids_created_list and jobs_created_cursor exist
+    created = []
+    rejected = []
 
-    @publish_event(Events.JOB_CREATED)
-    def publish_the_job_created(raw_job):
-        """Parse the job to a brewtils Job after replacing the unusable `id`."""
-        raw_object_id: ObjectId = raw_job.pop(id_key)
-        object_id_as_dict = json.loads(json_util.dumps(raw_object_id))
-        raw_job["id"] = object_id_as_dict[oid_key]
-
-        # the `publish_event` wrapper needs the whole Job object
-        return SchemaParser.parse_job(raw_job)
+    for job in jobs:
+        try:
+            created.append(create_job(job))
+        except (ModelValidationError, ValidationError) as exc:
+            rejected.append((job, str(exc)))
 
-    # but we're only interested in the `id` field
-    return list(
-        map(attrgetter("id"), map(publish_the_job_created, jobs_created_cursor))
-    )
+    return {"created": created, "rejected": rejected}
```

Comment:
This causes the wrapper that easy_client uses to enter into an infinite loop with an eventual *RecursionError* thrown. The function *serialize* has a warning in its docstring that

> This is potentially recursive

and maybe that wasn't such a good idea.

At any rate, and without jumping deeply into the code, I'm guessing the  brewtils schema needs to be updated and maybe this will go away...

Reply:
Very interesting.  I wouldn't expect there to be any interaction between this function and the easy_client, but I will work on reproducing what you found and get to the bottom of it.